src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -1131,4 +1168,34 @@ protected function runGitAction(string $status, GitHub $github, string $provider
             }
         }
     }
+
+    protected function listRules(Document $project, array $queries, Database $database, callable $callback = null): void
+    {
+        $cursor = null;
+
+        do {
+            $queries = \array_merge([
+                Query::limit(100),
+                Query::equal("projectInternalId", [$project->getInternalId()])
+            ], $queries);
+
+            if ($cursor !== null) {
+                $queries[] = Query::cursorAfter($cursor);
+            }
+
+            $results = $database->find('rules', $queries);
+
+            if (\count($results) > 0) {
+                $cursor = $results[\count($results) - 1];
+            } else {
+                $cursor = null;
```

Comment:
I think this will skip the last batch of the iteration 👍🏻 

instead of checking against count being `> 0` you should check if its less then you limit and then set it to null.

Reply:
oh yeah, you might be right actually.